app/models/grade_entry.rb

```diff
@@ -36,12 +34,14 @@ def attendance_perfect_earnings
 
   def math_improvement_earnings(previous_entry)
     return 0 unless previous_entry
+    return 0 if math_grade.nil? || previous_entry.math_grade.nil?
```

Comment:
Oh, it looks like I was wrong. 

When I tested, I ran into an issue when the grades were an empty string. 

After I save a grade like this: 
<img width="800" height="356" alt="image" src="https://github.com/user-attachments/assets/ef09654a-6db9-41f8-8669-1205cb00158b" />

student's math and reading are empty strings instead of nil.

So it should be `#blank?`, not `nil?`.